tests/unit/lib/reports/KinnexReport.spec.js

```diff
@@ -0,0 +1,377 @@
+        status: 200,
+        statusText: 'OK',
+        ok: true,
+        json: () => Promise.resolve({ data: [request], included: [sample] }),
+      })
+
+      const { data, errors } = await fetchTractionSamples('input', [])
+      expect(errors).toEqual({})
+      expect(data).toHaveLength(1)
+      expect(data[0]).toEqual({
+        request_id: request.id,
+        barcode: request.attributes.barcode,
+        cost_code: request.attributes.cost_code,
+        library_type: request.attributes.library_type,
+        external_study_id: request.attributes.external_study_id,
+        date_of_sample_collection: sample.attributes.date_of_sample_collection,
+        supplier_name: sample.attributes.supplier_name,
+        donor_id: sample.attributes.donor_id,
+        species: sample.attributes.species,
+        sanger_sample_id: sample.attributes.sanger_sample_id,
+      })
+    })
+
+    it('successful if only one of the requests returns data', async () => {
+      get.mockResolvedValueOnce({
+        status: 200,
+        statusText: 'OK',
+        ok: true,
+        json: () => Promise.resolve({ data: [request], included: [sample] }),
+      })
+      // Second call is for source_identifier fetch
+      get.mockResolvedValueOnce({
+        status: 200,
+        statusText: 'OK',
+        ok: true,
+        json: () => Promise.resolve({ data: [], included: [] }),
+      })
+
+      const { data, errors } = await fetchTractionSamples('input', [])
+      expect(errors).toEqual({})
+      expect(data).toHaveLength(1)
+      expect(data[0]).toEqual({
+        request_id: request.id,
+        barcode: request.attributes.barcode,
+        cost_code: request.attributes.cost_code,
+        library_type: request.attributes.library_type,
+        external_study_id: request.attributes.external_study_id,
+        date_of_sample_collection: sample.attributes.date_of_sample_collection,
+        supplier_name: sample.attributes.supplier_name,
+        donor_id: sample.attributes.donor_id,
+        species: sample.attributes.species,
+        sanger_sample_id: sample.attributes.sanger_sample_id,
+      })
+    })
+
+    it('fetches multiple samples from Traction API across filter types', async () => {
+      const request2 = pacbioRequestFactory.content.data[1]
+      const sample2 = pacbioRequestFactory.content.included[8]
+
+      get.mockResolvedValueOnce({
+        status: 200,
+        statusText: 'OK',
+        ok: true,
+        json: () => Promise.resolve({ data: [request], included: [sample] }),
+      })
+      // Second call is for source_identifier fetch
+      get.mockResolvedValueOnce({
+        status: 200,
+        statusText: 'OK',
+        ok: true,
+        json: () => Promise.resolve({ data: [request2], included: [sample2] }),
+      })
+      const { data, errors } = await fetchTractionSamples('sample1,sample2', [])
+      expect(errors).toEqual({})
+      expect(data).toEqual([
+        {
+          request_id: request.id,
```

Comment:
as above. Moving this into an object would significantly reduce no of lines, complexity and reusability.